app/src/main/java/com/termux/x11/MainActivity.java

```diff
@@ -176,7 +176,23 @@ protected void onCreate(Bundle savedInstanceState) {
         mInputHandler = new TouchInputHandler(this, new RenderStub.NullStub() {
             @Override
             public void swipeDown() {
-                toggleExtraKeys();
+                if (SamsungDexUtils.checkDeXEnabled(getApplicationContext())) {
+                    lorieView.releasePointerCapture();
+                    lorieParent.releasePointerCapture();
+                } else {
+                    toggleExtraKeys();
+                }
+            }
+            @Override
+            public void swipeUp() {
+                // Gesture for soft touchpad in DeX mode that cannot trigger KEYCODE_BACK easily
+                if (SamsungDexUtils.checkDeXEnabled(getApplicationContext())) {
+                    toggleKeyboardVisibility(MainActivity.this);
+                    Log.d("MainActivity", "Toggling keyboard visibility");
+                    if(inputMethodManager != null) {
+                        inputMethodManager.showSoftInput(lorieView, InputMethodManager.SHOW_FORCED);
```

Comment:
Is there a reason to trigger toggling IME twice?

Reply:
During my previous test, `toggleKeyboardVisibility` did not work. However, I cannot reproduce it now, so I think one `toggleKeyboardVisibility` should just work. 

I was mistaken in previous test perhaps because the keyboard toggling feature disappear sometimes when the app switches from phone screen to external display or vice versa. It isn't too annoying so now I simply close and re-open everytime.